src/scales/scale.time.js

```diff
@@ -176,6 +176,19 @@ module.exports = function(Chart) {
 			Chart.Scale.prototype.initialize.call(this);
 		},
 
+        /**
+         * Allow x-axis to be accessed via x or t attribute
+         */
+		getRightValue: function(rawValue) {
+			if (typeof(rawValue) === 'object') {
+				if ((rawValue instanceof Date) || rawValue.isValid) {
+					return rawValue;
+				}
+				return this.getRightValue(this.isHorizontal() ? (!helpers.isNullOrUndef(rawValue.t) ? rawValue.t : rawValue.x) : rawValue.y);
```

Comment:
I think we should fallback to x/y only if `t === undefined`. Also `t` should be considered for vertical scale:

```javascript
this.getRightValue(helpers.valueOrDefault(rawValue.t, this.isHorizontal() ? rawValue.x : rawValue.y));
```